src/Shared/ErrorUtilities.cs

```diff
@@ -52,9 +52,9 @@ public static void DebugTraceMessage(string category, string formatstring, param
 
         internal static void VerifyThrowInternalError(bool condition, string message, params object[] args)
         {
-            if (s_throwExceptions && !condition)
+            if (!condition)
```

Comment:
cleaned up this file 
1. remove the actual exception throwing from within VerifyThrowXX. this makes it more likely they can inline. "outside" the condition is hot.
2. add string resource verify to just outside every condition, and only there. inside the condition, it's going to try to load it anyway.
3. move s_throwExceptions inside the cold ThrowXX methods as well.